packetbeat starts logging errors with too many open files watching processes with -beta4

With 1.0.0-beta4, RHEL  7.1, 4 socket, 40 CPUs, 256 GB of memory ...

When I looked the open files, and this one packetbeat instance has close to 500+ /proc files opened.  Here are the logs from /var/log/message from one second:

```
2015-10-24T03:42:58.248673+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:288: FindSocketsOfPid: Open: open /proc/58955/fd: too many open files
2015-10-24T03:42:58.249177+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:288: FindSocketsOfPid: Open: open /proc/58956/fd: too many open files
2015-10-24T03:42:58.249551+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:288: FindSocketsOfPid: Open: open /proc/58957/fd: too many open files
2015-10-24T03:42:58.256024+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.266074+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.276254+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.287356+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.297990+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.308760+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.319017+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:153: Error finding PID files for httpd: Open /proc: open /proc: too many open files
2015-10-24T03:42:58.319460+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.330209+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.341199+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.351879+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.362335+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.373360+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.383770+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.394951+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.406156+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.417313+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.427626+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.439464+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.450666+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.461742+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.472272+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.483278+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.493968+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.504809+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.515655+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.526464+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.536990+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.546974+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.557704+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.568254+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.578937+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.589710+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.600668+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.611526+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.622117+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.632873+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.643688+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.654402+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.665359+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.676022+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.686598+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.697121+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.707641+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.718433+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.729224+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.740018+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.750833+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.761442+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.772413+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.783143+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.794491+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.805493+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.816312+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.826804+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.837874+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.849814+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.859699+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.870234+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.881108+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.892319+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.902868+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
2015-10-24T03:42:58.918103+00:00 perf34 /usr/bin/packetbeat[44016]: procs.go:271: Open: open /proc/net/tcp: too many open files
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,7 @@ All notable changes to this project will be documented in this file based on the
 ### Backward Compatibility Breaks
 
 ### Bugfixes
+- Close file descriptors used to monitor processes. #337
 
 ### Added
 

--- a/procs/procs.go
+++ b/procs/procs.go
@@ -168,6 +168,7 @@ func FindPidsByCmdlineGrep(prefix string, process string) ([]int, error) {
 	if err != nil {
 		return pids, fmt.Errorf("Open /proc: %s", err)
 	}
+	defer proc.Close()
 
 	names, err := proc.Readdirnames(0)
 	if err != nil {
@@ -271,6 +272,7 @@ func (proc *ProcessesWatcher) UpdateMap() {
 		logp.Err("Open: %s", err)
 		return
 	}
+	defer file.Close()
 	socks, err := Parse_Proc_Net_Tcp(file)
 	if err != nil {
 		logp.Err("Parse_Proc_Net_Tcp: %s", err)
@@ -365,6 +367,7 @@ func FindSocketsOfPid(prefix string, pid int) (inodes []int64, err error) {
 	if err != nil {
 		return []int64{}, fmt.Errorf("Open: %s", err)
 	}
+	defer procfs.Close()
 	names, err := procfs.Readdirnames(0)
 	if err != nil {
 		return []int64{}, fmt.Errorf("Readdirnames: %s", err)

